src/main/java/soot/jimple/toolkits/scalar/CopyPropagator.java

```
@@ -218,7 +218,10 @@ protected void internalTransform(Body b, String phaseName, Map<String, String> o
                 if (l != m) {
                   Integer defCount = localToDefCount.get(m);
                   if (defCount == null || defCount == 0) {
-                    throw new RuntimeException("Variable " + m + " used without definition!");
+                    if (Options.v().verbose()) {
```

Don't just ignore errors. If we generate invalid code, we need to know that. Keep in mind that Soot can also write out DEX code again. If we have wrong code, the output will be invalid as well. These sanity checks are highly important.

---

OK.I will close the PR.

---

I print the code of the error
Exception in thread "main" java.lang.RuntimeException: Undefined variable $u-1 in method: <com.android.server.policy.PhoneWindowManager$SwitchKeyboardLayoutMessageObject: int hashCode()>
Class: com.android.server.policy.PhoneWindowManager$SwitchKeyboardLayoutMessageObject

Method body:
   1: this := @this: com.android.server.policy.PhoneWindowManager$SwitchKeyboardLayoutMessageObject
   2: this.<android.apex.ApexInfo: boolean isFactory> = $u0
   3: $u0 = $u0
   4: $u0 = $u-1
   5: return $u0